app/src/main/java/app/revanced/manager/ui/component/patches/OptionFields.kt

```diff
@@ -91,29 +88,45 @@ private class OptionEditorScope<T : Any>(
     val option: Option<T>,
     val openDialog: () -> Unit,
     val dismissDialog: () -> Unit,
+    val selectionWarningEnabled: Boolean,
+    val showSelectionWarning: () -> Unit,
     val value: T?,
-    val setValue: (T?) -> Unit,
+    val setValue: (T?) -> Unit
 ) {
     fun submitDialog(value: T?) {
         setValue(value)
         dismissDialog()
     }
 
-    fun clickAction() = editor.clickAction(this)
+    fun showSelectionWarningDialogElse(block: () -> Unit) {
+        if (!option.required && selectionWarningEnabled)
+            showSelectionWarning()
+        else
+            block()
+    }
+
+    fun clickAction() {
+        showSelectionWarningDialogElse {
+            editor.clickAction(this)
+        }
+    }
 
     @Composable
     fun ListItemTrailingContent() = editor.ListItemTrailingContent(this)
 
     @Composable
     fun Dialog() = editor.Dialog(this)
+
+    @Composable
+    fun SelectionWarningDialog(onDismissDialog: () -> Unit) = SelectionWarningDialog(onDismiss = onDismissDialog)
```

Comment:
You can remove this function from `OptionEditorScope` and call it directly instead.